src/ontodia/data/sparql/sparqlDataProvider.ts

```diff
@@ -189,17 +206,36 @@ export class SparqlDataProvider implements DataProvider {
         elementIds: string[];
         linkTypeIds: string[];
     }): Promise<LinkModel[]> {
-        const ids = params.elementIds.map(escapeIri).map(id => ` ( ${id} )`).join(' ');
+        const elementIds = params.elementIds.filter(id => !BlankNodes.isEncodedBlank(id));
+
+        const additionalData = (
+            this.options.acceptBlankNodes ?
+            BlankNodes.linksInfo(params.elementIds) :
+            null
+        );
+
+        if (elementIds.length === 0 && this.options.acceptBlankNodes) {
+            return Promise.resolve(getLinksInfo(additionalData));
+        }
+
+        const ids = elementIds.map(escapeIri).map(id => ` ( ${id} )`).join(' ');
         const linksInfoQuery = resolveTemplate(this.settings.linksInfoQuery, {ids: ids});
         const query = this.settings.defaultPrefix + linksInfoQuery;
-        return this.executeSparqlQuery<LinkBinding>(query).then(getLinksInfo);
+        return this.executeSparqlQuery<LinkBinding>(query, {additionalData}).then(getLinksInfo);
     }
 
     linkTypesOf(params: { elementId: string; }): Promise<LinkCount[]> {
+        if (this.options.acceptBlankNodes && BlankNodes.isEncodedBlank(params.elementId)) {
+            return Promise.resolve(getLinksTypesOf(BlankNodes.linkTypesOf(params)));
+        }
         const elementIri = escapeIri(params.elementId);
         const query = this.settings.defaultPrefix
             + resolveTemplate(this.settings.linkTypesOfQuery, {elementIri: elementIri});
-        return this.executeSparqlQuery<LinkCountBinding>(query).then(getLinksTypesOf);
+        return this.executeSparqlQuery<LinkCountBinding>(query).
+            then(result => {
+                result.results.bindings = result.results.bindings.filter(b => !isRdfBlank(b.link));
```

Comment:
Add `.filter()` call to `getLinksTypesOf()` instead of modifying `bindings` here.